diff --git a/src/testdir/test_search.vim b/src/testdir/test_search.vim
--- a/src/testdir/test_search.vim
+++ b/src/testdir/test_search.vim
@@ -471,6 +471,10 @@ func Test_search_cmdline_incsearch_highl
   if !exists('+incsearch') || !has('terminal') || has('gui_running')
     return
   endif
+  if has('win32')
+    " This test doesn't work well on Windows. Disable it temporarily.
+    return
+  endif
   let h = winheight(0)
   if h < 3
     return
