From 01ad5d7804175987e87ed8aae32a31e4a774e189 Mon Sep 17 00:00:00 2001
From: "K.Takata" <kentkt@csc.jp>
Date: Tue, 2 Oct 2018 15:20:04 +0900
Subject: [PATCH 14/24] nsis: Fix some !ifdef conditions

---
 nsis/gvim.nsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/nsis/gvim.nsi b/nsis/gvim.nsi
index a51b385e5..bfb078779 100644
--- a/nsis/gvim.nsi
+++ b/nsis/gvim.nsi
@@ -36,6 +36,9 @@ Unicode true
 # Comment the next line if you do not want to add Native Language Support
 !define HAVE_NLS
 
+# Comment the next line if you do not want to include VisVim extension:
+!define HAVE_VIS_VIM
+
 # Comment the following line to create a multilanguage installer:
 !define HAVE_MULTI_LANG
 
@@ -587,6 +590,7 @@ SectionGroup $(str_group_plugin) id_group_plugin
 SectionGroupEnd
 
 ##########################################################
+!ifdef HAVE_VIS_VIM
 Section "$(str_section_vis_vim)" id_section_visvim
 	SectionIn 3
 
@@ -594,6 +598,7 @@ Section "$(str_section_vis_vim)" id_section_visvim
 	!insertmacro UpgradeDLL "${VIMSRC}\VisVim\VisVim.dll" "$0\VisVim.dll" "$0"
 	File ${VIMSRC}\VisVim\README_VisVim.txt
 SectionEnd
+!endif
 
 ##########################################################
 !ifdef HAVE_NLS
@@ -673,14 +678,18 @@ Section -post
 	  SectionGetSize ${id_section_editwith} $4
 	  IntOp $3 $3 + $4
 	${EndIf}
+!ifdef HAVE_VIS_VIM
 	${If} ${SectionIsSelected} ${id_section_visvim}
 	  SectionGetSize ${id_section_visvim} $4
 	  IntOp $3 $3 + $4
 	${EndIf}
+!endif
+!ifdef HAVE_NLS
 	${If} ${SectionIsSelected} ${id_section_nls}
 	  SectionGetSize ${id_section_nls} $4
 	  IntOp $3 $3 + $4
 	${EndIf}
+!endif
 
 	# Register EstimatedSize.
 	# Other information will be set by the install.exe.
@@ -832,11 +841,13 @@ Section "un.$(str_unsection_register)" id_unsection_register
 	# created.  Thus the "vim61" directory is included in it.
 	StrCpy $0 "$INSTDIR"
 
+!ifdef HAVE_VIS_VIM
 	# If VisVim was installed, unregister the DLL.
 	${If} ${FileExists} "$0\VisVim.dll"
 	  nsExec::Exec "regsvr32.exe /u /s $0\VisVim.dll"
 	  Pop $3
 	${EndIf}
+!endif
 
 	# delete the context menu entry and batch files
 	nsExec::Exec "$0\uninstal.exe -nsis"
@@ -907,7 +918,9 @@ Section "un.$(str_unsection_exe)" id_unsection_exe
 	RMDir /r $0\syntax
 	RMDir /r $0\tools
 	RMDir /r $0\tutor
+!ifdef HAVE_VIS_VIM
 	RMDir /r $0\VisVim
+!endif
 	RMDir /r $0\lang
 	RMDir /r $0\keymap
 	Delete $0\*.exe
-- 
2.17.0

