From 0a33244da95775bfa13317350d17eb6c6d0dcac1 Mon Sep 17 00:00:00 2001
From: "K.Takata" <kentkt@csc.jp>
Date: Sat, 13 Oct 2018 13:17:00 +0900
Subject: [PATCH 30/30] nsis: Include danish.nsi

---
 nsis/gvim.nsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/nsis/gvim.nsi b/nsis/gvim.nsi
index a4cd27ccb..ad504853c 100644
--- a/nsis/gvim.nsi
+++ b/nsis/gvim.nsi
@@ -158,6 +158,7 @@ Page custom SetCustom ValidateCustom
 
 # Include support for other languages:
 !ifdef HAVE_MULTI_LANG
+    !include "lang\danish.nsi"
     !include "lang\dutch.nsi"
     !include "lang\german.nsi"
     !include "lang\italian.nsi"
-- 
2.17.0

