From 9b09e87471d8d37e57c79617041bf17cad6bc5ae Mon Sep 17 00:00:00 2001
From: "K.Takata" <kentkt@csc.jp>
Date: Mon, 8 Oct 2018 18:37:31 +0900
Subject: [PATCH 31/33] nsis: Update Filelist

---
 Filelist | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Filelist b/Filelist
index 3052d7523..da19e3f2e 100644
--- a/Filelist
+++ b/Filelist
@@ -475,6 +475,7 @@ SRC_DOS =	\
 		nsis/gvim_version.nsh \
 		nsis/vimrc.ini \
 		nsis/README.txt \
+		nsis/lang/*.nsi \
 		uninstal.txt \
 		src/VisVim/Commands.cpp \
 		src/VisVim/Commands.h \
@@ -515,6 +516,7 @@ SRC_DOS_BIN =	\
 		src/xpm/x86/lib-vc14/libXpm.lib \
 		nsis/icons/*.bmp \
 		nsis/icons/*.ico \
+		nsis/icons/*.svg \
 
 # source files for Amiga, DOS, etc. (also in the extra archive)
 SRC_AMI_DOS =	\
-- 
2.17.0

