From 77f32422d420adf31e1c3c343bb9f2dd4e89b0b8 Mon Sep 17 00:00:00 2001
From: scootergrisen <scootergrisen@gmail.com>
Date: Fri, 26 Oct 2018 19:06:59 +0200
Subject: [PATCH 34/35] Update danish.nsi

---
 nsis/lang/danish.nsi | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/nsis/lang/danish.nsi b/nsis/lang/danish.nsi
index 16b9074a3..7b218fa2d 100644
--- a/nsis/lang/danish.nsi
+++ b/nsis/lang/danish.nsi
@@ -166,13 +166,13 @@ LangString str_msg_compat_title    ${LANG_DANISH}     " Vi- / Vim-opførsel "
 LangString str_msg_compat_desc     ${LANG_DANISH}     "&Kompatibilitet og forbedringer"
 LangString str_msg_compat_vi       ${LANG_DANISH}     "Vi-kompatibel"
 LangString str_msg_compat_vim      ${LANG_DANISH}     "Vim original"
-LangString str_msg_compat_defaults ${LANG_DANISH}     "Vim med nogle forbedringer"
-LangString str_msg_compat_all      ${LANG_DANISH}     "Vim med alle forbedringer (standard)"
+LangString str_msg_compat_defaults ${LANG_DANISH}     "Vim med nogle forbedringer (indlæs defaults.vim)"
+LangString str_msg_compat_all      ${LANG_DANISH}     "Vim med alle forbedringer (indlæs vimrc_example.vim) (standard)"
 
 LangString str_msg_keymap_title   ${LANG_DANISH}     " Tilknytninger "
 LangString str_msg_keymap_desc    ${LANG_DANISH}     "&Gentilknyt nogle få taster (Ctrl-V, Ctrl-C, Ctrl-A, Ctrl-S, Ctrl-F osv.)"
 LangString str_msg_keymap_default ${LANG_DANISH}     "Gentilknyt ikke taster (standard)"
-LangString str_msg_keymap_windows ${LANG_DANISH}     "Gentilknyt nogle få taster"
+LangString str_msg_keymap_windows ${LANG_DANISH}     "Gentilknyt nogle få taster til Windows"
 
 LangString str_msg_mouse_title   ${LANG_DANISH}     " Mus "
 LangString str_msg_mouse_desc    ${LANG_DANISH}     "&Opførsel af højre og venstre knapper"
-- 
2.17.0

