diff --git a/src/testdir/test_channel.vim b/src/testdir/test_channel.vim
index 6ad9dc0e1..04633bfdb 100644
--- a/src/testdir/test_channel.vim
+++ b/src/testdir/test_channel.vim
@@ -2466,7 +2466,7 @@ func LspOtCb(chan, msg)
 endfunc
 
 func LspTests(port)
-  " call ch_logfile('Xlsprpc.log', 'w')
+  call ch_logfile('Xlsprpc.log', 'w')
   let ch = ch_open(s:localhost .. a:port, #{mode: 'lsp', callback: 'LspCb'})
   if ch_status(ch) == "fail"
     call assert_report("Can't open the lsp channel")
@@ -2659,11 +2659,20 @@ func LspTests(port)
   call assert_fails('call ch_evalexpr(ch, [1, 2, 3])', 'E1206:')
   call assert_fails('call ch_sendexpr(ch, "abc")', 'E1206:')
   call assert_fails('call ch_evalexpr(ch, #{method: "ping"}, #{callback: "LspOtCb"})', 'E917:')
-  " call ch_logfile('', 'w')
+  call ch_logfile('', 'w')
 endfunc
 
 func Test_channel_lsp_mode()
   call RunServer('test_channel_lsp.py', 'LspTests', [])
+  if filereadable('Xlsprpc.log')
+    let s1 = "<-- Contents of Xlsprpc.log -->"
+    let s1 ..= readfile('Xlsprpc.log')->string()
+    let s1 ..= "<-- Contents of Xlspdebug.log -->"
+    let s1 ..= readfile('Xlspdebug.log')->string()
+    call assert_report(s1)
+    call delete('Xlsprpc.log')
+    call delete('Xlspdebug.log')
+  endif
 endfunc
 
 " vim: shiftwidth=2 sts=2 expandtab
diff --git a/src/testdir/test_channel_lsp.py b/src/testdir/test_channel_lsp.py
index 530258d84..f901398a9 100644
--- a/src/testdir/test_channel_lsp.py
+++ b/src/testdir/test_channel_lsp.py
@@ -228,7 +228,7 @@ class ThreadedTCPRequestHandler(socketserver.BaseRequestHandler):
 
     def handle(self):
         print("=== socket opened ===")
-        self.debug = False
+        self.debug = True
         msgbuf = ''
         while True:
             try:
